week_3/week_3_quiz.md

```diff
@@ -0,0 +1,134 @@
+
+- 3.11.4
+- 3.11.5
+- 3.11.6
+- 3.11.7
+
+**Question 8**
+
+8) Use docker commands to download the `bids/base_fsl` image [from Docker Hub](https://hub.docker.com/r/bids/base_fsl/). Also download [this map from NeuroVault.org](http://neurovault.org/media/images/457/tfMRI_SOCIAL_TOM-RANDOM_zstat1.nii.gz) and place it in an example folder in your current working directory. Using the `-v` option as described in the lecture (or “volume flags” if you are referencing [the Docker documentation](https://docs.docker.com/storage/volumes/)), mount your folder so that it will be accessible from the container when you run it. Now use the `fslstats` command to output the mean entropy of non-zero voxels for the image you downloaded. Hint: Use `fslstats --help` to determine the option that will return the mean entropy of non-zero voxels. What is the mean entropy of the non-zero voxels in the downloaded image?
+
+- 0.410776
+- 0.929105
+- 0.405803
+- 0.914086
+
+**Question 9**
+
+Next, we'll smooth the downloaded image using a Gaussian kernel with the FSL command `fslmaths /data/tfMRI_SOCIAL_TOM-RANDOM_zstat1.nii.gz -kernel gauss 10 -fmean /data/smoothed.nii.gz`. Confirm that this command, which you issued inside of the Docker container, added the `smoothed.nii.gz` output file to the example folder on your host system. Use the same commands as before to calculate the mean entropy of the nonzero voxels in the smoothed image. What is it?
+
+- 0.705882
+- 0.701026
+- 0.694043
+- 0.698877
+
+**Question 10**
+
+Now we will use the aforesaid `bids/base_fsl` image as a base image for a custom Docker image. Using your favorite editor, save a new file called Dockerfile inside an empty folder. The first line of this file should be:
+
+```
+FROM bids/base_fsl
+```
+
+In the next lines in your Dockerfile, download your favorite scientific computing software. For example, if you wanted to add python, your Dockerfile could look like:
+
+```
+FROM bids/base_fsl
+RUN wget http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O /opt/miniconda.sh
+RUN /bin/bash /opt/miniconda.sh -b -p /opt/conda
+ENV PATH="/opt/conda/bin:$PATH"
+```
+
+Build this docker image by navigating to the same directory as your Dockerfile and running `docker build -t my_custom_fsl .` Don't forget about the dot at the end, which tells Docker where to find the Dockerfile. Next register for an account on Docker Hub. Log into Docker Hub in your console using `docker login`. Build your image with the name `<username>/my_custom_fsl` and push your image to Docker Hub using `docker push <username>/my_custom_fsl.` Find a link to your image on Docker Hub and paste below
+
+---
+<input type="text" id="name" name="name"/>
+---
```

Comment:
I see the break in the rendered file but this doesn't seem to render the way you want.

Reply:
Thanks. I fixed this in the latest commit and chose to get rid of the text box entirely.